.github/workflows/ubuntu.yml

```diff
@@ -47,20 +50,64 @@ jobs:
       - name: collect code coverage
         run: bash <(curl -s https://codecov.io/bash) || echo "Codecov did not collect coverage reports"
 
-          #      - name: configure-format
-          #        run: |
-          #          clang-format --version
-          #          cmake -G Ninja -S test -B build-format -DCMAKE_FORMAT_PROGRAM=$(which clang-format-13)
-          #
-          #      - name: check style
-          #        run: |
-          #          cmake --build build-format --target check-format -v
+  format:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: actions/cache@v2
+        with:
+          path: "**/cpm_modules"
+          key: ${{ github.workflow }}-cpm-modules-${{ hashFiles('**/CMakeLists.txt', '**/*.cmake') }}
+
+      - uses: seanmiddleditch/gha-setup-ninja@master
+        with:
+          version: 1.10.2
+
+      - name: Install llvm ${{ env.LLVM_VER }}
+        run: |
+          sudo .github/workflows/llvm.sh ${LLVM_VER}
+          pip3 install cmakelang pyyaml
+
+      - name: configure-format
+        run: |
+          clang-format-${LLVM_VER} --version
+          cmake -G Ninja -S test -B build-format -DCLANG_FORMAT_PROGRAM=$(which clang-format-${LLVM_VER})
+
+      - name: display format diff
+        run: |
+          cmake --build build-format --target format -v
+
+      - name: check style
+        run: |
+          cmake --build build-format --target check-format -v
+
+  lint:
```

Comment:
Do you think it makes sense for these to be completely new checkouts as opposed to separate steps on the same checkout?

Reply:
They run in parallel and display better as separate jobs imo.

I guess it doesn't matter too much, `lint` will end up dominating